doc/whats_new/v0.23.rst

```diff
@@ -550,6 +550,14 @@ Miscellaneous
   error message is raised when y was expected but None was passed.
   :pr:`16622` by `Nicolas Hug`_.
 
+- |API| The default setting `print_changed_only` has been changed from False
+  to True. This means that the `repr` of estimators is now more concise and
+  only shows the parameters whose default value has been changed when
+  printing an estimator. You can restore the previous behaviour by using
+  `sklearn.set_config(print_changed_only=False)`. Also, note that it is
+  always possible to quickly inspect the parameters of any estimator using
+  `est.get_params()`. :pr:`17061` by `Nicolas Hug`_.
```

Comment:
Yes, I was looking to see if you had these docs. Nicely done. I'd consider noting deep=False to not overwhelm uses.